lib/codit/population/networks/home_locations.py

```diff
@@ -94,46 +123,66 @@ def allocate_households_to_each_building(list_types_average_households, list_coo
+            df_result = pd.concat([df_result, df_temp])
 
     return df_result.values.tolist()
 
 
-def build_households_home_list(total_h=50000):
+def allocate_homes_to_wards(total_h, coords_per_ward):
     """
-    Build a list of total_h households: ['lon', 'lat', 'building_type']
-    :param total_h: total number of households
-    :return: a full list of ['lon', 'lat', 'building_type']
+    Allocate homes to wards
+    :param total_h: total number of households in the Ward
+    :param coords_per_ward: The coordinates list of all accommodation buildings ['lon', 'lat', 'building_type', 'ward_name', 'ward_code'] in the Ward
+    :return: a full list of households in a Ward with info ['lon', 'lat', 'building_type', 'ward_name', 'ward_code']
     """
-    coords_types = get_coords(COORDINATES_CSV)
-    df_types_constraints_households = generate_average_number_homes_for_building_type(total_h, coords_types)
+    logging.info(
+        f"number of household in this ward is {total_h}, number of coordinates in this ward is {len(coords_per_ward)}")
+    df_types_constraints_households = generate_average_number_homes_for_building_type(total_h, coords_per_ward)
     list_types_average_households = list(
         zip(df_types_constraints_households['building_type'], df_types_constraints_households['number'],
             df_types_constraints_households['average_num_households'] - df_types_constraints_households[
                 'min_households'],
             df_types_constraints_households['min_households']))
-    list_num_households_per_building = allocate_households_to_each_building(list_types_average_households, coords_types)
+    list_num_households_per_building = allocate_households_to_each_building(list_types_average_households,
+                                                                            coords_per_ward)
 
     list_households_info = []
     for num_households_per_building in list_num_households_per_building:
-        if int(num_households_per_building[3]) > 0:
-            list_households_info += [num_households_per_building[:3]] * int(num_households_per_building[3])
-    df_home_list = pd.DataFrame(list_households_info, columns=['lon', 'lat', 'building_type'])
+        if int(num_households_per_building[5]) > 0:
+            list_households_info += [num_households_per_building[:5]] * int(num_households_per_building[5])
+    return list_households_info
+
+
+def build_households_home_list():
+    """
+    Build a full list of households: ['lon', 'lat', 'building_type','ward_name', 'ward_code']
+    :return: a full list of ['lon', 'lat', 'building_type', 'ward_name', 'ward_code']
+    """
+    coords_types = get_coords(COORDINATES_WARDS_CSV)
+    population_wards = get_population_wards(POPULATION_WARDS_CSV)
+    list_households_info = []
+    for pop_ward in population_wards:
+        tmp_coords_ward = []
+        for coord_type in coords_types:
+            if coord_type[4] == pop_ward[0]:
```

Comment:
`coord_type[4]` can be given a name, so this code can be better read and understood later